### PR TITLE
Jira Server: show integrated task 

### LINKED ISSUE #5994: Jira Server: show integrated task
Requires:
* #5810 
* #5993 

## Acceptance criteria
* [ ] call `createTaskIntegration` from the `TaskIntegrationMenu` for Jira Server
* [ ] add Jira Server to [`integration`](https://github.com/ParabolInc/parabol/blob/6893c722d92f4a3b0daed839c0b0e9aa977f2f4d/packages/server/graphql/types/Task.ts#L78) field on `Task`
* [ ] adapt UI to show integrated task

### PR DESCRIPTION
**Pull request by igorlesnenko**
Resolves #5994

Needs #6059

image

**How to test:**

- Add Jira Server integration
- Create a task in parabol
- Using push to integration menu, push a task to the selected Jira Server
- See the task was pushed successfully
- Open task in Jira Server, see it was created correctly
- Edit task in Jira Server
- Open parabol board, see the task is loaded correctly

### CODE REVIEW

**Comment by blimmer** [comment]
@igorlesnenko , I cherry-picked https://github.com/ParabolInc/parabol/pull/6104 into this branch to unblock you. Just FYI in case you see merge conflicts from it.

File: packages/server/dataloader/jiraServerLoaders.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  id: string
+  self: string
+  key: string
+  descriptionHTML: string
+  summary: string
+  description: string | null
+  project: {
+    key: string
+  }
+  type: 'JiraServer'
```
Comment (Dschoordsch) [inline packages/server/dataloader/jiraServerLoaders.ts:23]: -1 we should keep the spelling of `IntegrationProviderServiceEnum` 'jiraServer'

**Review by Dschoordsch — CHANGES_REQUESTED** [review]

File: packages/server/dataloader/jiraServerLoaders.ts (hunk 1 above)
Comment (igorlesnenko) [reply packages/server/dataloader/jiraServerLoaders.ts]: @Dschoordsch fixed

File: packages/client/components/TaskIntegrationLink.tsx [hunk 2]
```diff
@@ -44,6 +44,20 @@ const TaskIntegrationLink = (props: Props) => {
...
       </JiraIssueLink>
     )
+  } else if (integration.__typename === 'JiraServerIssue') {
+    const {url, key, projectKey} = integration
+    return (
+      <StyledLink
+        href={url}
+        rel='noopener noreferrer'
+        target='_blank'
+        title={`GitHub Issue #${key} on ${projectKey}`}
```
Comment (Dschoordsch) [inline packages/client/components/TaskIntegrationLink.tsx:54]: -1 It should be `Jira Server Issue #${key} ...`

File: packages/server/dataloader/jiraServerLoaders.ts [hunk 3]
```diff
@@ -14,6 +36,47 @@ export type JiraServerProject = JiraServerRestProject & {
...
 }
 
+export const jiraServerIssue = (
+  parent: RootDataLoader
+): DataLoader<JiraServerIssueKey, JiraServerIssue | null, string> => {
+  return new DataLoader<JiraServerIssueKey, JiraServerIssue | null, string>(
+    async (keys) => {
+      const results = await Promise.allSettled(
+        keys.map(async ({teamId, userId, issueId}) => {
+          const manager = await TaskIntegrationManagerFactory.initManager(parent, 'jiraServer', {
```
Comment (Dschoordsch) [inline packages/server/dataloader/jiraServerLoaders.ts:46]: -1 I think here we should create `JiraTaskIntegrationManager` directly. This also avoids dataloader calls manager which calls dataloader scenario.

File: packages/server/dataloader/jiraServerLoaders.ts [hunk 4]
```diff
@@ -14,6 +36,47 @@ export type JiraServerProject = JiraServerRestProject & {
...
+            sendToSentry(issueRes, {userId, tags: {issueId, teamId}})
+            return null
+          }
+          return {
+            id: issueRes.id,
+            self: issueRes.self,
+            key: issueRes.key,
+            descriptionHTML: issueRes.renderedFields.description,
+            ...issueRes.fields,
+            type: 'jiraServer'
```
Comment (Dschoordsch) [inline packages/server/dataloader/jiraServerLoaders.ts:67]: +1 we call this `service` everywhere else

File: packages/server/graphql/types/JiraServerIssue.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+const JiraServerIssue = new GraphQLObjectType<any, GQLContext>({
+  name: 'JiraServerIssue',
+  description: 'The Jira Issue that comes direct from Jira Server',
+  interfaces: () => [TaskIntegration],
+  isTypeOf: ({type}) => type === 'jiraServer',
+  fields: () => ({
+    id: {
+      type: new GraphQLNonNull(GraphQLID)
+    },
+    key: {
```
Comment (Dschoordsch) [inline packages/server/graphql/types/JiraServerIssue.ts:19]: +1 I would prefer calling it `issueKey` to match `JiraIssue`

File: packages/server/dataloader/jiraServerLoaders.ts [hunk 6]
```diff
@@ -14,6 +36,47 @@ export type JiraServerProject = JiraServerRestProject & {
   providerId: number
 }
 
+export const jiraServerIssue = (
```
Comment (Dschoordsch) [inline packages/server/dataloader/jiraServerLoaders.ts:39]: +2 Nice that you created a dataloader directly

File: packages/server/integrations/TaskIntegrationManagerFactory.ts [hunk 7]
```diff
@@ -40,11 +41,13 @@ export interface TaskIntegrationManager {
...
     integrationHash: string
   )
+
+  getIssue?(taskId: string)
 }
 
 export default class TaskIntegrationManagerFactory {
   public static async initManager(
-    dataLoader: DataLoaderWorker,
+    dataLoader: DataLoaderWorker | RootDataLoader,
```
Comment (Dschoordsch) [inline packages/server/integrations/TaskIntegrationManagerFactory.ts:50]: -2 this looks wrong, but gets resolved if you don't call this factory from the dataloader anymore

**Review by Dschoordsch — CHANGES_REQUESTED** [review]

**Comment by igorlesnenko** [comment]
merged into #6059 as easier to test and do changes

**Pull request merged**